tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -1724,13 +1724,39 @@ class DynamicBroadcastInDimOpNotActuallyDynamic
                                 PatternRewriter& rewriter) const override {
     auto type = op.getType().dyn_cast<RankedTensorType>();
     auto operandType = op.operand().getType().dyn_cast<RankedTensorType>();
-    if (!type || !type.hasStaticShape() || !operandType ||
+    auto outputDimOp = op.output_dimensions().getDefiningOp();
+    auto outputDimType =
+        op.output_dimensions().getType().dyn_cast<RankedTensorType>();
+    if (!type || !operandType || !outputDimType ||
         !operandType.hasStaticShape()) {
-      return rewriter.notifyMatchFailure(op, "requires static shape");
+      return rewriter.notifyMatchFailure(op, "requires operand static shape");
     }
-    rewriter.replaceOpWithNewOp<BroadcastInDimOp>(
-        op, op.getType(), op.operand(), op.broadcast_dimensions());
-    return success();
+    if (type.hasStaticShape()) {
```

Comment:
Can you add a comment documenting the pattern you're catching and how they are replaced?